spark_log_parser/eventlog.py

```diff
@@ -1,27 +1,22 @@
 import json
 import tempfile
 from pathlib import Path
-from urllib.parse import ParseResult
 
 import pandas as pd
 
-from spark_log_parser.extractor import Extractor, ExtractThresholds
-
 
 class EventLogBuilder:
     def __init__(
         self,
-        source_url: ParseResult | str,
+        event_log_paths: list[Path] | list[str],
         work_dir: Path | str,
-        s3_client=None,
-        extract_thresholds=ExtractThresholds(),
     ):
-        self.source_url = source_url
+
+        self.event_log_paths = self._validate_event_log_paths(event_log_paths)
         self.work_dir = self._validate_work_dir(work_dir)
-        self.s3_client = s3_client
-        self.extractor = Extractor(
-            self.source_url, self.work_dir, self.s3_client, extract_thresholds
-        )
+
+    def _validate_event_log_paths(self, event_log_paths: Path | str) -> Path:
```

Comment:
I think you want list[Path] | list[str] and list[Path] here

Reply:
Best to fix in now :)